ci-operator/step-registry/fips-check/node-scan/fips-check-node-scan-commands.sh

```diff
@@ -0,0 +1,133 @@
+        echo "setting the proxy"
+        echo "source ${SHARED_DIR}/proxy-conf.sh"
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "no proxy setting."
+    fi
+}
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+set_proxy
+run_command "oc whoami"
+run_command "oc version -o yaml"
+
+# skip for ARM64
+mapfile -t node_archs< <(oc get nodes -o jsonpath --template '{range .items[*]}{.status.nodeInfo.architecture}{"\n"}{end}' | sort -R | uniq -d)
+echo "The architectures included in the current cluster nodes are ${node_archs[*]}"
+NON_SUPPORTED_ARCHES=(arm64)
+for arch in "${node_archs[@]}"; do
+  if [[ "${NON_SUPPORTED_ARCHES[*]}" =~ $arch ]]; then
+      echo "Skip this test since it doesn't support $arch."
+      exit 0
+  fi
+done
+
+# get a master node
+pass=false
+master_node_0=`oc get node -l node-role.kubernetes.io/master= --no-headers | grep -Ev "NotReady|SchedulingDisabled"| awk '{print $1}' | awk 'NR==1{print}'`
+if [[ -z $master_node_0 ]]; then
+    echo "Error master node0 name is null!\n."
+    exit
+fi
+# create a ns
+project="node-scan-$RANDOM"
+run_command "oc new-project $project --skip-config-write"
+if [ $? == 0 ]; then
+    echo "create $project project successfully" 
+else
+    echo "fail to create $project project."
+fi
+# check whether it is disconnected cluster
+oc label namespace "$project" security.openshift.io/scc.podSecurityLabelSync=false pod-security.kubernetes.io/enforce=privileged pod-security.kubernetes.io/audit=privileged pod-security.kubernetes.io/warn=privileged --overwrite=true
+http_proxy=`oc get proxy cluster -o=jsonpath='{.spec.httpProxy}'`
+https_proxy=`oc get proxy cluster -o=jsonpath='{.spec.httpProxy}'`
+attempt=0
+if [ "$http_proxy" != "" ] && [ "$https_proxy" != "" ]; then
+    while true; do
+        out=$(oc --request-timeout=60s -n "$project" debug node/"$master_node_0" -- chroot /host bash -c "export http_proxy=$http_proxy;export https_proxy=$https_proxy; curl -I ifconfig.me --connect-timeout 5")
+        if [[ $out == *"HTTP"* ]]; then
+            echo "This is not a disconnected cluser"
+            break
+        fi
+        attempt=$(( attempt + 1 ))
+        if [[ $attempt -gt 3 ]]; then
+            echo "This is a disconnected cluser, skip testing"
+            oc delete ns "$project"
+            exit 0
+        fi
+        sleep 5
+    done
+else
+    while true; do
+        out=$(oc --request-timeout=60s -n "$project" debug node/"$master_node_0" -- chroot /host bash -c "curl -I ifconfig.me --connect-timeout 5")
+        if [[ $out == *"HTTP"* ]]; then
+            echo "This is not a disconnected cluser"
+            break
+        fi
+        attempt=$(( attempt + 1 ))
+        if [[ $attempt -gt 3 ]]; then
+            echo "This is a disconnected cluser, skip testing"
+            oc delete ns "$project"
+            exit 0
+        fi
+        sleep 5
+    done
+fi
```

Comment:
If you like, below is more concise to avoid duplicate lines:
```suggestion
attempt=0
while true; do
    out=$(oc --request-timeout=60s exec -n openshift-kube-apiserver kube-apiserver-"$master_node_0" -- bash -c 'http_proxy=$HTTP_PROXY curl -sSI ifconfig.me --connect-timeout 5')
    if [[ $out == *"HTTP"* ]]; then
        echo "This is not a disconnected cluser"
        break
    fi
    attempt=$(( attempt + 1 ))
    if [[ $attempt -gt 3 ]]; then
        echo "This is a disconnected cluser, skip testing"
        oc delete ns "$project"
        exit 0
    fi
    sleep 5
done
```
Note use single quote in `bash -c`. It works in both proxy and non-proxy env. In proxy env, core components automatically have the proxy vars:
```
$ out=$(oc --request-timeout=60s exec -n openshift-kube-apiserver kube-apiserver-"$master_node_0" -- bash -c 'env | grep -i proxy; http_proxy=$HTTP_PROXY curl -sSI ifconfig.me --connect-timeout 5')

$ echo "$out"
HTTP_PROXY=http://proxy-user1:xxx@xxx:3128
NO_PROXY=.cluster.local,.svc,10.0.0.0/16,10.128.0.0/14,127.0.0.1,172.30.0.0/16,api-int.xxx.com,localhost,test.no-proxy.com
HTTPS_PROXY=https://proxy-user1:xxx@xxx:3130

HTTP/1.1 ...
```

Reply:
If you open ifconfig.me in browser, it shows insecure http service, not https service, so it does not care $https_proxy env var. And why using http_proxy=$HTTP_PROXY, it is because `man curl`: `The environment variables can be specified in lower case or upper case.  http_proxy is an exception as it is only available in lower case.`

Reply:
Others LGTM.

Reply:
Thanks a lot for your feedback. I am not quite sure about the below command. I will have a try tomorrow and give you feedback.
`    out=$(oc --request-timeout=60s exec -n openshift-kube-apiserver kube-apiserver-"$master_node_0" -- bash -c 'http_proxy=$HTTP_PROXY curl -sSI ifconfig.me --connect-timeout 5')`